source/common/http/http1/balsa_parser.cc

```diff
@@ -193,21 +194,28 @@ size_t BalsaParser::execute(const char* slice, int len) {
       }
     }
 
-    if (message_type_ == MessageType::Request && !allow_custom_methods_ &&
-        !isFirstCharacterOfValidMethod(*slice)) {
-      status_ = ParserStatus::Error;
-      error_message_ = "HPE_INVALID_METHOD";
-      return 0;
+    if (message_type_ == MessageType::Request && !allow_custom_methods_) {
+      // Skip `(\r|\n).*`.
+      while (allow_cr_and_lf_at_request_start_ && len > 0 && (*slice == '\r' || *slice == '\n')) {
+        slice += 1;
+        skipped_bytes += 1;
+        len -= 1;
+      }
+      if (len > 0 && !isFirstCharacterOfValidMethod(*slice)) {
```

Comment:
what happens if we just processed an entire buffer of \r or \n?? Seems like we'll set `first_byte_processed_` even though no new request byte was processed and we'd run into the state of `HPE_INVALID_EOF_STATE`?

Reply:
Good catch. I believe this should fix.